### PR TITLE
Add new variable for flexibility between project id & name

### PR DESCRIPTION
**Pull request by sly92**
Is this change independant enough ?

### CODE REVIEW

**Review by ludoo — CHANGES_REQUESTED** [review]
we never use random ids in our modules for a variety of reasons, let's please keep this PR focuse don splitting id and name

**Comment by sly92** [comment]
Ok I understand

**Comment by ludoo** [comment]
Thanks for the change. I would much prefer not changing the main variable used for project id, as that would require changing every end to end example that uses the project module.

I would just add an optional `descriptive_name` variable, and use that to set the project name if not null.

**Comment by sly92** [comment]
That is what I thought. Thanks I will do it

File: modules/project/main.tf [hunk 1]
```diff
@@ -85,7 +88,7 @@ resource "google_project" "project" {
   org_id              = local.parent_type == "organizations" ? local.parent_id : null
   folder_id           = local.parent_type == "folders" ? local.parent_id : null
   project_id          = "${local.prefix}${var.name}"
-  name                = "${local.prefix}${var.name}"
+  name                = "${local.prefix}${local.name}"
```
Comment (ludoo) [inline modules/project/main.tf:91]: I would skip the prefix if user passes in `descriptive_name`.

File: modules/project/outputs.tf [hunk 2]
```diff
@@ -16,7 +16,7 @@
 
 output "project_id" {
   description = "Project id."
-  value       = "${local.prefix}${var.name}"
+  value       = "${local.prefix}${local.temp_project_id}"
```
Comment (ludoo) [inline modules/project/outputs.tf:19]: This should not be changed as project ID will stay the same as before this PR.

**Review by ludoo — CHANGES_REQUESTED** [review]

**Review by sly92 — COMMENTED** [review]
Adding new local variable to allow having "project name" différent from the "project id"

**Comment by sly92** [comment]
Do you know why the check failed ?

**Comment by sly92** [comment]
yes but what specifically ? 
do you need me to change something?

**Comment by juliocc** [comment]
The README.md for the project module is out-of date. You have to run the tfdoc script (in the tools folder) to update variables and ouputs.

**Comment by sly92** [comment]
I've just created a CLA. 
Do you need me to push again ?

**Review by ludoo — APPROVED** [review]

**Comment by sly92** [comment]
Having said that, thanks for all your helps

**Comment by ludoo** [comment]
@sly92 it also needs a CLA fro your company address (sla*****.ext@or*****.com), apparently.

**Comment by ludoo** [comment]
Ignore, the CLA bot is asleep, all looks good on our side. Thanks a lot for your contribution!

**Comment by sly92** [comment]
Sorry the useless space, can I revert and correct this ?

**Comment by ludoo** [comment]
what would you want to correct?

**Comment by sly92** [comment]
just remove the useless linebreak

**Pull request merged**